Tests have now their own database, and the database is properly regenerated at the initial state before each test

jocel1 opened the pull request:
<!-- Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information: -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Improve unit tests
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | composer test \o/

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/8379)
<!-- Reviewable:end -->

jocel1 committed:
CO: add SQL comments

jocel1 committed:
TE: Tests have now their own database, and the database is properly regenerated at the initial state before each tests.

jocel1 commented:
WIP, now tests need to be fixed!

jocel1 committed:
Add a few more fixes to tests

Quetzacoalt91 commented:
Hello @jocel1,

You should work on the branch `develop`, as the 1.7.2.3 should be the last version released of the `1.7.2.x` will go live in the next few weeks (days?).

jocel1 commented:
no problem, I'll rebase it on develop!

jocel1 committed:
TE: add a few more fixes

jocel1 committed:
TE: fix phpunit-admin tests

jocel1 committed:
TE: remove wrong teardown

jocel1 committed:
TE: more fixes

jocel1 committed:
TE: fix tests

jocel1 committed:
TE: more fixes

jocel1 committed:
TE: fix tests

jocel1 committed:
TE: fix tests

jocel1 committed:
TE: redirect error / warnings to /dev/null

jocel1 committed:
TE: fix typo

jocel1 committed:
TE: fix wrong command

jocel1 committed:
TE: fix all tests

jocel1 committed:
TE: reuse databaseDump class

xBorderie commented on src/PrestaShopBundle/Install/Install.php:
```diff
@@ -293,18 +324,129 @@ public function clearDatabase($truncate = false)
+    public static function restoreTestDB()
+    {
+        if (!file_exists(sys_get_temp_dir().'/'.'ps_dump.sql')) {
+            throw new DBALException('You need to run composer create-test-db to create the initial test database');
```
Maybe use `"You need to run 'composer create-test-db' to create the initial test database"`?

xBorderie commented:
As far as I can tell, there was only one added strings, for which I made a suggestion. What do you think, @AlexEven ?

jocel1 committed:
TE: wording

jocel1 committed:
TE: do not generate .htaccess file. Properly set PS_BASE_URI

mickaelandrieu commented:
I like it, I need to find time to review this one :+1:  Good job @jocel1 !

mickaelandrieu commented on classes/DateRange.php:
```diff
@@ -57,7 +57,7 @@ public static function getCurrentRange()
 		SELECT `id_date_range`, `time_end`
 		FROM `'._DB_PREFIX_.'date_range`
-		WHERE `time_end` = (SELECT MAX(`time_end`) FROM `'._DB_PREFIX_.'date_range`)');
+		WHERE `time_end` ORDER BY time_end DESC');
```
why do you need to change this SQL query?

jocel1 replied:
not needed at all, it was a buggy pending optimizations not related to this PR.

mickaelandrieu commented on install-dev/init.php:
```diff
@@ -25,6 +25,33 @@
+// setting the memory limit to 128M only if current is lower
+$memory_limit = ini_get('memory_limit');
+if (substr($memory_limit, -1) != 'G'
+    and ((substr($memory_limit, -1) == 'M' and substr($memory_limit, 0, -1) < 128)
```
can you please use ``&&`` and `||` operators instead? is this file loaded even in "non-test" mode?

jocel1 replied:
oh shit, cut and paste is bad

jocel1 replied:
and yes it's loaded even in non test mode in install script, but it actually fix potential issues

mickaelandrieu commented on src/PrestaShopBundle/Install/Install.php:
```diff
@@ -293,18 +324,130 @@ public function clearDatabase($truncate = false)
+            }
+        }
+    }
+
```
remove this empty line please :)

mickaelandrieu reviewed (dismissed):
Good job! Some minor requests

jocel1 committed:
TE: post review fixes. Add a few extra mocks for context

eternoendless commented:
Reviewed 13 of 18 files at r1, 2 of 6 files at r2, 22 of 24 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[classes/DateRange.php, line 60 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvanihnKmzlNEFTkRip:-KvanihoV7v5ODdgXW8p:bv3lrjb) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/classes/DateRange.php#L60)):*
> ```PHP
> 		SELECT `id_date_range`, `time_end`
> 		FROM `'._DB_PREFIX_.'date_range`
> 		WHERE `time_end` ORDER BY time_end DESC');
> ```

Maybe add `LIMIT 1`?

---

*[classes/Tools.php, line 53 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvaoCxnXxUIT7zDxovZ:-KvaoCxnXxUIT7zDxov_:b-9g7ech) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/classes/Tools.php#L53)):*
> ```PHP
>     }
> 
>     public static function resetRequest()
> ```

Please add a PhpDoc header on your new methods.

It is important that each method describes:
- A short summary of what it does
- Type (and if possible, a description) of each parameter
- Type and description of what it returns
- Exceptions thrown and when

If your method is an extension of a parent class' method, you can use `@inheritdoc`.

Documenting methods is important because
- It enables autocompletion by IDEs
- It eases code understanding

Example:
```php
   /**
     * Executes the current command.
     *
     * This method is not abstract because you can use this class
     * as a concrete class. In this case, instead of defining the
     * execute() method, you set the code to execute by passing
     * a Closure to the setCode() method.
     *
     * @param InputInterface  $input  An InputInterface instance
     * @param OutputInterface $output An OutputInterface instance
     *
     * @return null|int null or 0 if everything went fine, or an error code
     *
     * @throws LogicException When this abstract method is not implemented
     *
     * @see setCode()
     */
    protected function execute(InputInterface $input, OutputInterface $output)
    {
        // ...
    }
```

You can find out more about PhpDoc by reading the [proposed PSR-5 standard](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md).

---

*[classes/Tools.php, line 2139 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvaoP1lJ6FxtY2rhJmV:-KvaoP1lJ6FxtY2rhJmW:bo1g588) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/classes/Tools.php#L2139)):*
> ```PHP
>     {
>         if (defined('_PS_IN_TEST_')
>             || (defined('PS_INSTALLATION_IN_PROGRESS') && $rewrite_settings === null)) {
> ```

When using multiline ifs, please put the closing parenthesis in a new line, like this:

```php
if ($foo
    && $bar
) {
    // do something
}
```

---

*[src/PrestaShopBundle/Install/DatabaseDump.php, line 122 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbCch1KASb6Oas9H1Q:-KvbCch1KASb6Oas9H1R:b6na468) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/DatabaseDump.php#L122)):*
> ```PHP
>     {
>         $dumpCommand = $this->buildMySQLCommand('mysqldump', array($this->databaseName));
>         $dumpCommand .= ' > ' . escapeshellarg($this->dumpFile). ' 2> /dev/null';
> ```

Why are you suppressing stderr?

---

*[src/PrestaShopBundle/Install/Install.php, line 305 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbDNYgiN0PLoJXyjPU:-KvbDNYgiN0PLoJXyjPV:bvdr67o) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L305)):*
> ```PHP
>     public function generateSf2ProductionEnv()
>     {
>         $schemaUpgrade = new UpgradeDatabase(defined('_PS_IN_TEST_')?'test':null);
> ```

Please add a space around the `?` and `:` operators

---

*[src/PrestaShopBundle/Install/Install.php, line 360 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbE6ruYOM1Lsc6G13f:-KvbE6ruYOM1Lsc6G13g:b-jbu0u9) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L360)):*
> ```PHP
> 
> 
>     public function initializeTestContext()
> ```

Missing Phpdoc

---

*[src/PrestaShopBundle/Install/Install.php, line 407 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbE8_hu5lEA1W2O6V6:-KvbE8_iM_nFenXQX_Lw:b-jbu0u9) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L407)):*
> ```PHP
>     }
> 
>     public static function createTestDB() {
> ```

Missing Phpdoc

---

*[src/PrestaShopBundle/Install/Install.php, line 407 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbFVt4Crz-QKktObpe:-KvbFVt4Crz-QKktObpf:bkrhusd) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L407)):*
> ```PHP
>     }
> 
>     public static function createTestDB() {
> ```

I don't think this method belongs here. To me, this looks like a fixture that _consumes_ `Install`, not a part of it.

---

*[src/PrestaShopBundle/Install/Install.php, line 441 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbEA2wDNfRWCu4wvJw:-KvbEA2wDNfRWCu4wvJx:b-jbu0u9) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L441)):*
> ```PHP
>     }
> 
>     public static function restoreTestDB()
> ```

Missing Phpdoc

---

*[src/PrestaShopBundle/Install/Install.php, line 441 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbFkAfjCKO6jhZBVi9:-KvbFkAfjCKO6jhZBViA:b2xn095) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L441)):*
> ```PHP
>     }
> 
>     public static function restoreTestDB()
> ```

Same comment as `restoreTestDB()`

---

*[src/PrestaShopBundle/Install/Install.php, line 480 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbFwv4N7Fu63jEyCZx:-KvbFwv4N7Fu63jEyCZy:b6ro94i) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L480)):*
> ```PHP
>                     }
>                 }
>                 $iso_codes_to_install = array_flip(array_flip($iso_codes_to_install));
> ```

:thinking: this is odd...
Should use `array_unique` instead maybe?

---

*[src/PrestaShopBundle/Tests/Model/Product/AdminModelAdapterTest.php, line 172 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbGn9AGrzy7Vc3i6qi:-KvbGn9AGrzy7Vc3i6qj:bn2mbwf) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Tests/Model/Product/AdminModelAdapterTest.php#L172)):*
> ```PHP
>         \Context::getContext()->shop = new \Shop(1);
>         \Context::getContext()->language = new \Language(1);
>         \Context::getContext()->currency = new \Currency(1);
> ```

Why not use `ContextMocker`?
You may need to rebase on develop

---

*[tests/bootstrap-admin.php, line 28 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbHwvWTa-yngUzFgVZ:-KvbHwvWTa-yngUzFgV_:b93l3yv) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/bootstrap-admin.php#L28)):*
> ```PHP
> define('_PS_ROOT_DIR_', __DIR__ . '/..');
> define('_PS_IN_TEST_', true);
> umask(0000); // This will let the permissions be 0775
> ```

Shouldn't this be `002` for that?

---

*[tests/Integration/Core/Module/HookRepositoryTest.php, line 54 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJEGP8sU8bOo8nn8C:-KvbJEGP8sU8bOo8nn8D:b-a0le91) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Integration/Core/Module/HookRepositoryTest.php#L54)):*
> ```PHP
>         $modules = [
>             'ps_emailsubscription',
>             'ps_featuredproducts'
> ```

Why did you change the module here?

---

*[tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php, line 53 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJxM8AmWzB5C8wjJm:-KvbJxM8AmWzB5C8wjJn:bj68i9t) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php#L53)):*
> ```PHP
> <form>
>     <input type='text' name='card_number'>
> <button style='display:none' id='pay-with-OPTION_ID' type='submit'></button>
> ```

Looks like you removed an indentation here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

eternoendless reviewed (changes requested):
See my comments on reviewable :)

jocel1 committed:
TE: post review fixes

jocel1 commented:
Review status: 34 of 38 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[src/PrestaShopBundle/Install/DatabaseDump.php, line 122 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbCch1KASb6Oas9H1Q:-KvbOhriFgKOOn-OBGO1:b-ermapi) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/DatabaseDump.php#L122)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Why are you suppressing stderr?
</blockquote></details>

Because the command is generating a warning in Travis about password used in command line, and some tests are failing because of this warning

---

*[src/PrestaShopBundle/Install/Install.php, line 407 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbFVt4Crz-QKktObpe:-KvbTzvxYH18-vv5fEcL:bdz1t15) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L407)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

I don't think this method belongs here. To me, this looks like a fixture that _consumes_ `Install`, not a part of it.
</blockquote></details>

I agree, moved into PrestaShopBundle\Tests\Utils\Database

---

*[src/PrestaShopBundle/Install/Install.php, line 480 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbFwv4N7Fu63jEyCZx:-KvbW4evSHoo6cdghWGO:b-1fqat2) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Install/Install.php#L480)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

:thinking: this is odd...
Should use `array_unique` instead maybe?
</blockquote></details>

Changed :) It seems the reason for this code is: https://stackoverflow.com/questions/8321620/array-unique-vs-array-flip

---

*[src/PrestaShopBundle/Tests/Model/Product/AdminModelAdapterTest.php, line 172 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbGn9AGrzy7Vc3i6qi:-KvbWQM5j9QhP6NNLYbD:b-ms1d30) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/src/PrestaShopBundle/Tests/Model/Product/AdminModelAdapterTest.php#L172)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Why not use `ContextMocker`?
You may need to rebase on develop
</blockquote></details>

at this point of the code, since the objects are available, it was easier to use real objects instead of mocks

---

*[tests/Integration/Core/Module/HookRepositoryTest.php, line 54 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJEGP8sU8bOo8nn8C:-KvbYgNENhAe4NZFqOPh:b-1bwzlt) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Integration/Core/Module/HookRepositoryTest.php#L54)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Why did you change the module here?
</blockquote></details>

because ps_shoppingcart is not part of the modules we install during the test installation. List of available modules is quite short actually: https://github.com/PrestaShop/PrestaShop/tree/develop/tests/resources/modules

---

*[tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php, line 53 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJxM8AmWzB5C8wjJm:-KvbZ5Bok8C3C_4BY7jA:bcalk8i) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php#L53)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Looks like you removed an indentation here
</blockquote></details>

It's expected, the test was failing with the extra indentation

---

*[classes/DateRange.php, line 60 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvanihnKmzlNEFTkRip:-KvbZBxJuv-7e7RB5N1r:bzcx591) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/classes/DateRange.php#L60)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Maybe add `LIMIT 1`?
</blockquote></details>

I reverted the change, not related to this PR

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

jocel1 commented:
Reviewed 1 of 24 files at r3.
Review status: 28 of 41 files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

jocel1 commented:
Reviewed 9 of 18 files at r1, 1 of 6 files at r2, 19 of 24 files at r3, 3 of 5 files at r5, 11 of 11 files at r6.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

jocel1 commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[tests/bootstrap-admin.php, line 28 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbHwvWTa-yngUzFgVZ:-KvbcHTA7he34w_uLVyC:b-l2m8cy) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/bootstrap-admin.php#L28)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Shouldn't this be `002` for that?
</blockquote></details>

Comment changed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

Quetzacoalt91 commented on src/PrestaShopBundle/Install/Install.php:
```diff
@@ -293,18 +324,89 @@ public function clearDatabase($truncate = false)
+        $cldrUpdate->init();
+
+        //get each defined languages and fetch cldr datas
+        $langs = \DbCore::getInstance()->executeS('SELECT * FROM '._DB_PREFIX_.'lang');
```
Using classes with `Core` prevents all override to be used.

Please use `Db` instead, and declare it in the list of `use [...]` statements.

jocel1 replied:
I'm not sure we want to call override in case of an install?

Quetzacoalt91 replied:
This may not happen, but this issue on other classes made us work on the whole src/ folder to remove these `<class>Core` calls

Quetzacoalt91 commented on src/PrestaShopBundle/Install/Install.php:
```diff
@@ -293,18 +324,89 @@ public function clearDatabase($truncate = false)
+        $langs = \DbCore::getInstance()->executeS('SELECT * FROM '._DB_PREFIX_.'lang');
+
+        foreach ($langs as $lang) {
+            $cldrRepository = \Tools::getCldr(null, $lang['locale']);
```
Please declare `use Tools` at the top of the file.

Quetzacoalt91 commented on src/PrestaShopBundle/Install/Install.php:
```diff
@@ -293,18 +324,89 @@ public function clearDatabase($truncate = false)
+        }
+        if (!isset($context->currency) || !\Validate::isLoadedObject($context->currency)) {
+            if ($id_currency = (int)Configuration::get('PS_CURRENCY_DEFAULT')) {
+                $context->currency = new \Currency((int)$id_currency);
```
Same here, please declare your classes at the top of the file

```php
use Cache;
use Cookie;
use Currency;
use Validate;
```

jocel1 committed:
post review fixes

jocel1 commented:
changes done!

eternoendless commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php, line 53 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJxM8AmWzB5C8wjJm:-Kw9WyswH5JqWb_zOJZ0:b-10sn9w) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php#L53)):*
<details><summary><i>Previously, jocel1 (Jocelyn Fournier) wrote…</i></summary><blockquote>

It's expected, the test was failing with the extra indentation
</blockquote></details>

What? How come? If a test fails because of code indentation, there's something wrong somewhere 😐

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-:-Kw9YaEfSVnBKpdkLeOQ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

eternoendless commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php, line 53 at r4](https://reviewable.io:443/reviews/prestashop/prestashop/8379#-KvbJxM8AmWzB5C8wjJm:-KwF-rn4SUcQAHcc5XSz:b-vgisxi) ([raw file](https://github.com/prestashop/prestashop/blob/86f8c80622a1cac103fe5be21421278bbae8633a/tests/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php#L53)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

What? How come? If a test fails because of code indentation, there's something wrong somewhere 😐
</blockquote></details>

We'll fix that later

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8379)*
<!-- Sent from Reviewable.io -->

eternoendless commented:
Thank you @jocel1
